code/game/machinery/portable_turret.dm

```diff
@@ -98,109 +102,32 @@
 	req_access = list(ACCESS_CENT_SPECOPS)
 	one_access = FALSE
 
-/obj/machinery/porta_turret/proc/setup()
-	var/obj/item/gun/energy/E = new installation	//All energy-based weapons are applicable
-	var/obj/item/ammo_casing/shottype = E.ammo_type[1]
-
-	projectile = shottype.projectile_type
-	eprojectile = projectile
-	shot_sound = shottype.fire_sound
-	eshot_sound = shot_sound
-
-	weapon_setup(installation)
-
-/obj/machinery/porta_turret/proc/weapon_setup(guntype)
-	switch(guntype)
-		if(/obj/item/gun/energy/laser/practice)
-			lethal_is_configurable = FALSE
-			iconholder = 1
-			eprojectile = /obj/item/projectile/beam
-
-		if(/obj/item/gun/energy/laser/retro)
-			iconholder = 1
-
-		if(/obj/item/gun/energy/laser/captain)
-			iconholder = 1
-
-		if(/obj/item/gun/energy/lasercannon)
-			iconholder = 1
-
-		if(/obj/item/gun/energy/taser)
-			lethal_is_configurable = FALSE
-			eprojectile = /obj/item/projectile/beam
-			eshot_sound = 'sound/weapons/laser.ogg'
-
-		if(/obj/item/gun/energy/gun)
-			eprojectile = /obj/item/projectile/beam	//If it has, going to kill mode
-			eshot_sound = 'sound/weapons/laser.ogg'
-			egun = 1
-
-		if(/obj/item/gun/energy/gun/nuclear)
-			eprojectile = /obj/item/projectile/beam	//If it has, going to kill mode
-			eshot_sound = 'sound/weapons/laser.ogg'
-			egun = 1
-
-		if(/obj/item/gun/energy/gun/turret)
-			eprojectile = /obj/item/projectile/beam	//If it has, going to copypaste mode
-			eshot_sound = 'sound/weapons/laser.ogg'
-			egun = 1
+/obj/machinery/porta_turret/process()
+	if(!enabled || (stat & (NOPOWER|BROKEN)))
+		if(!always_up)
+			pop_down()
+		return
 
-		if(/obj/item/gun/energy/pulse/turret)
-			eprojectile = /obj/item/projectile/beam/pulse/hitscan
-			eshot_sound = 'sound/weapons/pulse.ogg'
-	if(initial_eprojectile)
-		eprojectile = initial_eprojectile
-	if(initial_projectile)
-		projectile = initial_projectile
+	if(!LAZYLEN(targets_being_processed))
+		return
 
-GLOBAL_LIST_EMPTY(turret_icons)
-/obj/machinery/porta_turret/update_icon_state()
-	if(!GLOB.turret_icons)
-		GLOB.turret_icons = list()
-		GLOB.turret_icons["open"] = image(icon, "openTurretCover")
+	if(process_targets(LAZYACCESS(targets_being_processed, TURRET_PRIMARY_TARGET), TURRET_PRIMARY_TARGET))
```

Comment:
I feel like the logic done here, should be instead down at the top of process_targets, so that you only need to input 1 instead of 2 copies of the same define
```suggestion
	if(process_targets(TURRET_PRIMARY_TARGET))
```